front/temporal/tracker/activities.ts

```diff
@@ -336,9 +335,27 @@ export const processTrackerNotificationWorkflowActivity = async ({
     return;
   }
 
-  // @todo Implement the notification logic here.
   localLogger.info(
-    `Processing tracker ${trackerId} with ${generations.length} generations.`
+    `[Tracker] Processing tracker ${trackerId} for workspace ${workspaceId}.`
+  );
+
+  const generations = tracker.generations || [];
+
+  if (generations.length) {
+    await sendTrackerEmailWithNoGeneration(tracker);
+  } else {
+    await sendTrackerEmailWithGenerations(tracker);
+  }
+
+  // Consume the tracker & associated generations.
+  await TrackerConfigurationResource.consumeGenerations({
+    auth,
+    trackerId,
+    generationIds: generations.map((g) => g.id),
```

Comment:
So here you might have notified about a generation that you aren't consuming here right ? since you aren't passing the generations to `sendTrackerEmailWithGenerations` ?

Reply:
Nope it's the same generations since all comes from tracker witch is a typed object. But I could make the code clearer. I'll add a commit

Reply:
ah yes of course ! makes senst

Reply:
I still added a commit to simplify the email function!